src/core/ddsc/include/dds/dds.h

```diff
@@ -976,13 +976,26 @@ dds_set_listener(dds_entity_t entity, const dds_listener_t * listener);
  * @ingroup domain_participant
  * @component participant
  *
- * If domain is set (not DDS_DOMAIN_DEFAULT) then it must match if the domain has also
- * been configured or an error status will be returned.
- * Currently only a single domain can be configured by providing configuration file.
- * If no configuration file exists, the default domain is configured as 0.
- *
- *
- * @param[in]  domain The domain in which to create the participant (can be DDS_DOMAIN_DEFAULT). DDS_DOMAIN_DEFAULT is for using the domain in the configuration.
+ * This function creates a new participant in the specified domain id, implicitly creating
+ * a domain entity if one doesn't exist for the specified domain id.  The domain entity
+ * can exist as a consequence of the existence of another participant in it, or because it
+ * was explicitly created using @ref dds_create_domain or @ref
+ * dds_create_domain_with_rawconfig.  If a domain entity is created, its configuration is
```

Comment:
"If a domain entity is created" -> "If the domain entity has not been created, a new domain entity is created using configuration taken from the CYCLONEDDS_URI environment variable". 

This might be easier to understand for noobs like me.